packages/utils/object.ts

```diff
@@ -15,3 +15,7 @@ export function checkObjectHasProperties(object: any, properties: string[]) {
 		if (!(prop in object)) throw new Error(`Missing property "${prop}": ${JSON.stringify(object)}`);
 	}
 }
+
+export const hasOwnProperty = (object: object, property: string): boolean => {
+	return !!object && Object.prototype.hasOwnProperty.call(object, property);
```

Comment:
Alternative: [`Object.hasOwn`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwn) (may require changing `"target": "es2017"` in `tsconfig.json` to `"target": "es2022"`).